src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -188,11 +188,14 @@ private async Task ProcessQueueAsync()
 
                     var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
 
+                    // The request context must be created serially inside the queue to so that requests always run
+                    // on the correct snapshot as of the last request.
+                    var context = await work.CreateRequestContextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
fix is here - the context must be created serially in the queue instead in `StartRequestAsync` which runs in parallel (non-mutating requests)